src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -41,6 +47,30 @@
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        SimpleDateFormat formatDate = new SimpleDateFormat("dd.MM.yyyy");
+        Date dateStart = formatDate.parse(dateFrom);
+        Date dateEnd = formatDate.parse(dateTo);
+        Date actualDate = null;
+        StringBuilder result = new StringBuilder("Отчёт за период " + dateFrom + " - " + dateTo);
+        int salary = 0;
+
+        if (dateStart.compareTo(dateEnd) >= 0) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+
+        for (String name : names) {
+            salary = 0;
+            for (String line : data) {
+                String[] entry = line.split(" ");
+                actualDate = formatDate.parse(entry[0]);
+                if (name.equals(entry[1])
+                        && actualDate.compareTo(dateStart) >= 0
+                        && actualDate.compareTo(dateEnd) <= 0) {
```

Comment:
it's better to use isBefore() or isAfter()

Reply:
Done. Thank you!